### DISCUSSION TITLE
Web UI wont get token

### DISCUSSION THREAD

**Discussion by moore** [tsightler/ring-mqtt#764] [Support Q&A]
I open the UI put it a working user name and password (tested by logging in to ring web site) hit the submit button and nothing happens and I see nothing in the logs. I have restarted the whold computer. I really have no idea where to look for another error. Al that is in my log is:

```
-------------------------------------------------------
| Ring-MQTT with Video Streaming                      |
| Addon for Home Assistant                            |
|                                                     |
| For support questions please visit:                 |
| https://github.com/tsightler/ring-mqtt/discussions  |
-------------------------------------------------------
cont-init: info: /etc/cont-init.d/ring-mqtt.sh exited 0
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun ring-mqtt (no readiness notification)
s6-rc: info: service legacy-services successfully started
-------------------------------------------------------
ring-mqtt.js version: 5.6.3
Node version v18.17.1
NPM version 9.1.2
git version 2.38.5
-------------------------------------------------------
Running ring-mqtt...
2023-12-06T06:15:20.479Z ring-mqtt Detected runmode: addon
2023-12-06T06:15:20.487Z ring-mqtt Configuration file: /data/options.json
2023-12-06T06:15:25.903Z ring-mqtt State file /data/ring-state.json not found. No saved state data available.
2023-12-06T06:15:25.904Z ring-mqtt No refresh token was found in state file, generate a token using the addon Web UI
2023-12-06T06:15:25.908Z ring-mqtt Succesfully started the token generator web UI
2023-12-06T06:15:25.935Z ring-mqtt Discovered MQTT Host: core-mosquitto
2023-12-06T06:15:25.937Z ring-mqtt Discovered MQTT Port: 1883
2023-12-06T06:15:25.939Z ring-mqtt Discovered MQTT User: addons
2023-12-06T06:15:25.942Z ring-mqtt Discovered MQTT password: <hidden>
2023-12-06T06:15:25.947Z ring-mqtt MQTT URL: mqtt://addons:********@core-mosquitto:1883
```

**Comment by moore** [comment]
It seems to be making the request to:
` http://homeassistant:8123/api/hassio_ingress/i22i9FqGEuArm_7teQH2x-XkvGfh9_VEifdNncTqeaE/submit-account`
but that never gets a response. (based on chrome web developer tools)

**Comment by dsp-avt** [comment]
What worked for me was to login to my ring account on the ring.com website and then go to Control Center -> Authorized Client Devices. From there delete the authorized home-assistant device. I also had to delete an "unknown device", which apparently was a failed attempt for the ring-mqtt add-on to get an access token. Once I had those devices deleted I was able to get to the 2FA screen in the ring-mqtt GUI.

**Comment by NathanT15** [comment]
I'm running into this exact issue trying to set up this add-on for the first time. The "Submit" button seems to be broken, nothing happens when clicked.

**Reply by tsightler** [reply]
The submit button is definitely not broken, there's barely anything that can even be broken about it as it it just a standard HTML form with a submit function that calls a URL.  This code is unchanged for years and there's almost nothing to it.  I just went through the entire process of removing the addon, deleting the authorized device from my Ring account, re-installing, and re-authenticating.  No issues at all.  Not to mention that this project adds around 15-20 users/week so if this was a widespread problem, it would be reported much more widely.

Of course, that's not saying you are not having a problem, only that the problem is almost certainly something unique to the specific setups that are having the issue.  Perhaps the Chrome version in use, or some plugin or anti-malware tool that is blocking the submit.  What happens if you just type a few random characters into the username and click submit?

Access to the addon website is proxied by HA via the internal "Ingress" feature, so it could be something with that, although this has never been an issue in years either and I just tested all of this in my setup with no issues.
